pallets/detach/src/copy_pasted_beefy_merkle_tree.rs

```diff
@@ -0,0 +1,144 @@
+//! Everything in this crate was copied from: https://crates.io/crates/beefy-merkle-tree/10.0.0 (which is also used in Substrate)
```

Comment:
less silly name please.
Can you simply seriously call it `merkle_tree.rs`? You explained the rest in comments...

Reply:
done